### DISCUSSION TITLE
updated hook does not work on a dynamic nested array

### DISCUSSION THREAD

**Discussion by schroderd** [livewire/livewire#8318] [7. Bugs (archived)]
### Livewire version

v.3.4.10

### Laravel version

v.10.48.7

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

In this code

```php
class NestedArrayBug extends Component
{
    public array $foo = ['bar' => []];

    public function updatedFooBar(): void
    {
        dd('Nothing to see');
    }

    public function updated($propertyName): void
    {
        dd($propertyName);
    }

    public function render(): string
    {
        return <<<'blade'
        <div>
            @for($i=0; $i<5; $i++)
                <label>
                    <input type="checkbox" wire:model.live="foo.bar" value="{{ $i }}">
                    {{ $i }}
                </label>
            @endfor
        </div>
        blade;
    }
}
```
updatedFooBar never gets called

### Repository or Wirebox URL

https://wirebox.app/b/4koq5

### Additional information

_updatedFooBar_ does not get called when a checkbox is clicked.
This seems to be due to the dynamic nature of the nested array in _bar_.
This code works fine on Livewire 2, but not with Livewire 3.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by akamasAT** [comment]
this will only work with updatedFoo()

**Reply by schonhoff** [reply]
That is correct. It would only trigger the updatedFoo hook. If it was not a dynamic/numeric array, it would work as well. For example:

```php
class NestedArrayBug extends Component
{
    public array $foo = ['bar' => ['work' => '']];

    public function updatedFooBarWork(): void
    {
        dd('Nothing to see');
    }

    public function updated($propertyName): void
    {
        dd($propertyName);
    }

    public function render(): string
    {
        return <<<'blade'
        <div>
            @for($i=0; $i<5; $i++)
                <label>
                    <input type="checkbox" wire:model.live="foo.bar" value="{{ $i }}">
                    {{ $i }}
                </label>
            @endfor
        </div>
        blade;
    }
}
```

The `updatedFooBarWork()` hook would be called. On a numeric array it would be the best to call the last assoc parents hook (like described above). It worked fine in Livewire 2 but the hook was removed in Livewire 3. If it is intentional or not I couldn't tell.

Our solution/workaround currently is to use the `updatedFoo()` hook and use an if statement with the condition `str($propertyName)->startsWith('foo.bar')` and call the method which would be called in the updatedFooBar method from Livewire 2.

**Comment by danie-ramdhani** [comment]
Hmm yeah... Probably bug because it's not checking array values and only checking the nested values, i've check it with `updatedFooBar0()`. Maybe you can submit a failing test PR for it...
